src/containers/FileDetailContainer.js

```diff
@@ -0,0 +1,66 @@
+import React, {useState} from 'react';
+import {withRouter} from 'react-router-dom';
+import FileDetail from '../components/FileDetail/FileDetail';
+import {LoadingPlaceholder} from '../components/Loading';
+import {Query, Mutation} from 'react-apollo';
+import {UPDATE_FILE} from '../state/mutations';
+import {GET_FILE_BY_ID} from '../state/queries';
+import PropTypes from 'prop-types';
+
+const FileDetailContainer = ({kfId, history, match}) => {
+  const [fileNameInput, setFileName] = useState();
+  // Updates the file then routes to the study's files listing
+  const onSubmit = (e, updateFile, description, fileType) => {
+    e.preventDefault();
+    const name = fileNameInput;
+    updateFile({variables: {kfId, name, description, fileType}})
+      .then(() => {
+        history.goBack();
+      })
+      .catch(err => console.log(err));
+  };
+
+  return (
+    <Query query={GET_FILE_BY_ID} variables={{kfId}}>
+      {({loading, error, data}) => {
+        if (loading) return <LoadingPlaceholder componentName="File Editor" />;
+        if (error) return `Error!: ${error}`;
+        if (fileNameInput === undefined) setFileName(data.fileByKfId.name);
+        return (
+          <Mutation
+            mutation={UPDATE_FILE}
+            refetchQueries={res => [
+              {
+                query: GET_FILE_BY_ID,
+                variables: {kfId},
+              },
+            ]}
+          >
+            {(updateFile, {_}) => {
+              return (
+                <FileDetail
+                  fileNode={data.fileByKfId}
+                  onSubmit={e =>
```

Comment:
The `onSubmit` prop name isn't very clear what it's doing. Maybe `onUpdate` or something would be better.